File: python/problem013.py
import urllib2
import re

def large_number_sum(first_n_digits):
    # read data from project-euler
    data = re.search(r'>\s\d(?s).*</div><b', 
           urllib2.urlopen('http://projecteuler.net/problem=13').read())
    lines = []
    pattern = re.compile(r'\d+')
    # data is still mixed with html now
    for char in data.group().splitlines():
        # purge data from html and add digits to the list
        match = re.search(pattern, char)
        if match:
            lines.append(int(match.group()))
    # return the first_n_digits of the big number sum
    return str(sum(lines))[:first_n_digits]

print large_number_sum(10)

File: ruby/problem013.rb
require 'net/http'

def large_sum(n_digits)
	uri = URI.parse("http://projecteuler.net/problem=13")
  html = Net::HTTP.get(uri)
  data = /(^[\d]+<br \/>\n)+/.match(html)[0]
  data = data.lines.each do |x|
  	x.gsub(/<br \/>|\n/, '')
  end
  # convert to int, sum and convert back to string
  data.collect!{ |x| x.to_i }.reduce(:+).to_s[0...n_digits]

end

puts large_sum(10)